### PR TITLE
Tooling: Adds support for Jetpack extensions & widgets to code signing lanes

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #19361 
Ref: pdnsEh-FT-p2

## Description
Updates our code signing lanes to handle Jetpack extensions and widgets. 

_P.S: The CI is expected to fail at first because the actual provisional profiles are yet to be created._

## Testing Instructions
No testing is needed. Once CI succeeds in building the Jetpack Installable Build, it means we're good. 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

6. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
I run this update automation after creating the App Identifiers and it worked fine 👍

**Comment by mokagio** [comment]
I pushed [a fix](https://github.com/wordpress-mobile/WordPress-iOS/pull/19384/commits/340052030acc662c9c13919db90fcd6e2233caac) for the CI issue. If CI passes, I'll merge this PR so `trunk` gets that fix, too, as it's something we experienced on Woo as well today.

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -24394,7 +24395,7 @@
 				OTHER_SWIFT_FLAGS = "$(inherited) -D COCOAPODS -D DEBUG";
 				PRODUCT_BUNDLE_IDENTIFIER = com.automattic.jetpack.JetpackShare;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "WordPress Share Extension Development";
+				PROVISIONING_PROFILE_SPECIFIER = "Jetpack iOS Development Share Extension";
 				SKIP_INSTALL = YES;
 				SWIFT_OBJC_BRIDGING_HEADER = "WordPressShareExtension/WordPressShare-Bridging-Header.h";
```
Comment (mokagio) [inline WordPress/WordPress.xcodeproj/project.pbxproj:24400]: I was surprised by this using WordPress at the start, then I realized that `JetpackShareExtension` is sort of a fake target. It uses all the same code as `WordPressShareExtension` except for a few different configurations that give it its own identity. I expect the two to diverge more in the future.

**Comment by hassaanelgarem** [comment]
> I pushed [a fix](https://github.com/wordpress-mobile/WordPress-iOS/pull/19384/commits/340052030acc662c9c13919db90fcd6e2233caac) for the CI issue. If CI passes, I'll merge this PR so trunk gets that fix, too, as it's something we experienced on Woo as well today.

@mokagio Just a heads up that this PR wasn't targeting `trunk`. I [created a PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/19409) with just the fix so it can be merged on `trunk`.

**Pull request merged**